src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java

```diff
@@ -29,6 +29,9 @@
         "threads.corePoolSize= 3",
         "threads.queueCapacity= 1",
         "threads.maxPoolSize= 4",
+        "subscription-handler.threads.corePoolSize= 3",
```

Comment:
Will this not be a bottleneck if EventHandler thread works faster than the SubscriptionHandler,, we will queue up more and more threads. So Subscriptionhandler threadpool should be much bigger than Eventhandler threadpool.
This could be a good load test scenario, that tests number SubscritionHandler threads needed for different EventHandler threadpool sizes.
What happens if threads.queueCapacity is reached, is those new threads never created and Aggregations never Subscription checked and no triggering is happening ?

Reply:
Hmm, I'm not sure if it will be a bottleneck (or if it's a big risk) . The problem we saw before was that an infinite amount of threads were created from SubscriptionHandler and because it reads from the database, it blocked all other processing of events (all EventHandler threads were blocked). Now at least we have a limit on the amount of threads allowed for subscription handler. It's a trade off, which one do we want to prioritize: processing events from rabbitmq or trigger subscriptions as fast as possible?

Reply:
Is it possible to do a "baton pass" so that you wait to process until a thread is available? But then I guess both operations they might as well be in the same thread process...

Reply:
I explained the situation in a chat.
Short summary of the situation I am trying to explain:
If EventHandler thread (event consumer) executes much faster than SubscriptionHandler that executes much slower due to MongoDb connection,queries, updates and read read of big JSON objects, then a lot of threads will be queuing up in SubscriptionHandler thread pool.
And what happens when that queue is full.
If RabbitMq queue has 100 000 events, than we almost must have 100 000 in queue capacity in SubscriptionHandler treadpool queue, to be on safe side so we don't loose any aggregation subscription checks that is made by SubscriptionHandler thread.

Reply:
If eventHandler threads blocks until a new thread slot is free in SubscriptionHandler thread queue, then it will not be a problem.

Reply:
I understand your point @tobiasake but it's hard to test such a scenario. For now I have the same amount of threads in both pools. And don't forget that the event handler also does read/write to mongodb to merge and update aggregations so I don't think it executes so much faster than subscription handler threads do.

Reply:
I guess it would use bigger event linked chains that causes bigger Aggregations and casuign longer/bigger opeations agains MongoDb
It also possible to mock som functions in code and add sleeps in subscriptionHandler threads,, so it takes longer time to process subscriptionhandler threads.

Reply:
I am mean, why implement something to fix a issue without testing that specific situation?

Reply:
It sounds complex, doesn't the thread pool have some sort of queue? This queue would keep growing though if the SubscriptionHandler is slower and more and more events are coming in.

Reply:
Of what I can understand on the info about ThreadPool,, queue throws exeception if queue is full. So Aggregation and matching subscription will not happen for those aggreagation that happens when subscriptionQueueHandler Queue is full. So alot of Subscripion triggering will never happen if its alot of events in MessageBus queue and EI consumens for event than number of SubscritpionHandler can handle due to the ThreadPool limitations.